app/pdfUpload/pdfService.js

```diff
@@ -53,12 +53,32 @@
             return filename;
         }
 
-        service.getReadableURL = function getReadableURL(url, callback) {
+        service.deleteFile = function deleteFile(fileURL) {
+            var deferred = $q.defer();
+            var storage = firebase.storage().refFromURL(fileURL);
+            if (storage) {
```

Comment:
Esse if é desnecessario pois a chamada do método refFromURL sempre retorna um objeto, ou seja independentemente do arquivo existir ou não no firebase ele retornará um objeto. O tratamento do erro só acontece de fato quando a promisse for executada, é ai onde vai ser verificado se o arquivo existe ou não.